instant-segment/src/lib.rs

```diff
@@ -77,13 +77,14 @@ impl Segmenter {
         &self,
         input: &str,
         search: &'a mut Search,
-    ) -> Result<impl ExactSizeIterator<Item = &'a str>, InvalidCharacter> {
+    ) -> Result<(impl ExactSizeIterator<Item = &'a str>, f64), InvalidCharacter> {
```

Comment:
I feel like the API might be both more elegant and backwards compatible if we make a concrete type (maybe `SegmentIter`) that implements `ExactSizeIterator` and also has a `score()` method. What do you think?

Reply:
actually I have found that the `impl ExactSizeIterator` is a little annoying - I had been considering submitting a pull request that instead just returned `&'a [String]`.

it feels like an unnecessary limitation for users that they can only consume the returned iterator exactly once and have to take a clone of their own if they want to do more than only iterate it.  A slice gives me more flexibility.

So I would prefer not to double down on the iterator design.

Reply:
I suppose my issue with that is it "leaks" the `smartstring::alias::String` instead of just yielding the `&str` part. What flexibility have you needed from the return type?

(In general, curious to hear what you're using instant-segment for.)

Reply:
The particular flexibility I want is mitigated by this pull request but eg suppose you want both to print the sentence and also get its score.  

`SmartString` is already "leaked" from the API on `Segmenter.new()` where the unigrams and bigrams inputs must contain them.  Perhaps if these had been `Into<String>`...?

I am afraid I am not using instant-segmenter for anything important.  Actually I am re-visiting the same silly puzzle I talked about at #53.